app/src/org/commcare/activities/connect/ConnectIdActivity.java

```diff
@@ -0,0 +1,166 @@
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_connect_id);
+        NavHostFragment host2 = (NavHostFragment)getSupportFragmentManager().findFragmentById(R.id.nav_host_fragment_connectid);
+        controller = host2.getNavController();
+        Bundle extras = getIntent().getExtras();
+
+        String value="";
+        if (extras != null) {
+            value = extras.getString("TASK");
+        }
+        switch (value){
+            case ConnectConstants.BIGIN_REGISTRATION -> beginRegistration1(this);
+            case ConnectConstants.UNLOCK_CONNECT -> unlockConnect(this);
+            case ConnectConstants.VERIFY_PHONE -> beginSecondaryPhoneVerification(this);
+        }
+    }
+
+//    @Override
+//    public void onBackPressed() {
+//
+//        int count = getSupportFragmentManager().getBackStackEntryCount();
+//
+//        if (count == 0) {
+//            super.onBackPressed();
+//            //additional code
+//        } else {
+//            getSupportFragmentManager().popBackStack();
+//        }
+//
+//    }
+
+    private ConnectIdBiometricConfigFragment getCurrentFragment() {
+        NavHostFragment navHostFragment =
+                (NavHostFragment) getSupportFragmentManager().findFragmentById(R.id.nav_host_fragment_connectid);
+        Fragment currentFragment =
+                navHostFragment.getChildFragmentManager().getPrimaryNavigationFragment();
+        if (currentFragment instanceof ConnectIdBiometricConfigFragment) {
+            return (ConnectIdBiometricConfigFragment) currentFragment;
+        }
+        return null;
+    }
+
+    public static void beginRegistration1(Context parent) {
+        forgotPassword = false;
+        forgotPin = false;
+        int requestCode = ConnectConstants.CONNECT_NO_ACTIVITY;
+        switch (ConnectManager.getStatus()) {
+            case NotIntroduced ->
+                    navDirections = org.commcare.fragments.connectId.ConnectIdRecoveryDecisionFragmentDirections.actionConnectidRecoveryDecisionSelf();
+            case Registering -> {
+                ConnectUserRecord user = ConnectDatabaseHelper.getUser(parent);
+                int phase = user.getRegistrationPhase();
+                if (phase != ConnectConstants.CONNECT_NO_ACTIVITY) {
+                    requestCode = phase;
+                } else if (user.shouldForcePin()) {
+                    navDirections = org.commcare.fragments.connectId.ConnectIdRecoveryDecisionFragmentDirections.actionConnectidRecoveryDecisionToConnectidPin(ConnectConstants.CONNECT_UNLOCK_PIN, user.getPrimaryPhone(), user.getPassword());
+                } else if (user.shouldForcePassword()) {
+                    navDirections = org.commcare.fragments.connectId.ConnectIdRecoveryDecisionFragmentDirections.actionConnectidRecoveryDecisionToConnectidPassword(ConnectConstants.CONNECT_UNLOCK_PASSWORD, user.getPrimaryPhone(), user.getPassword());
```

Comment:
Similar to Pin, if we're navigating to actionConnectidRecoveryDecisionToConnectidPassword why do we have to pass in CONNECT_UNLOCK_PASSWORD as well?